src/main/java/com/project/controller/AdminCommentController.java

```diff
@@ -0,0 +1,80 @@
+import com.project.model.request.RequestComment;
+import com.project.model.response.CommentResponse;
+import com.project.repository.ApplicationRepository;
+import com.project.service.AdminCommentService;
+import com.project.service.ApplicationService;
+import com.project.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@CrossOrigin(origins = "*")
+@RequestMapping("api/comments")
+public class AdminCommentController {
+    private final AdminCommentService commentService;
+    private final ApplicationService applicationService;
+    private final UserService userService;
+
+    @Autowired
+    public AdminCommentController(AdminCommentService commentService, ApplicationService applicationService, UserService userService){
+        this.applicationService = applicationService;
+        this.userService = userService;
+        this.commentService = commentService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<CommentResponse>> fetchComments() {
+        List<AdminComment> comments = commentService.getAll();
+        List<CommentResponse> response = new ArrayList<>();
+
+        for(AdminComment comment: comments) {
+            response.add(new CommentResponse(comment.getComment(), comment.getAuthor().getEmail(), comment.getCommentDate(), comment.getDateModified()));
+        }
+
+        return new ResponseEntity<>(response, HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<CommentResponse> fetchComment(@PathVariable Long id) throws AdminCommentNotFoundException {
+        AdminComment comment = commentService.getById(id);
+        return new ResponseEntity<>(
+                new CommentResponse(comment.getComment(), comment.getAuthor().getEmail(), comment.getCommentDate(), comment.getDateModified()),
+                HttpStatus.OK
+        );
+    }
+
+    @PostMapping
+    public ResponseEntity<HttpStatus> createComment(@Valid @RequestBody RequestComment comment) throws ApplicationNotFoundException, UserNotFoundException {
+        AdminComment adminComment = new AdminComment(comment.getComment(), comment.getDate(),
+                applicationService.getById(comment.getApplicationId()), userService.getById(comment.getAuthorId()));
+        commentService.addAdminComment(adminComment);
+        return new ResponseEntity<>(HttpStatus.CREATED);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<HttpStatus> updateComment(@RequestBody RequestComment comment, @PathVariable Long id)
+            throws AdminCommentNotFoundException, ApplicationNotFoundException, UserNotFoundException {
+        AdminComment adminComment = new AdminComment(comment.getComment(), comment.getDate(),
```

Comment:
`AdminComment` sounds very specific. I would suggest creating an Interface or an Abstract class Comment and then extend or implement them for `AdminComment` since if in the future you will need to have let's say `UserComment` you will just need to extend or implement `Comment` and add User-specific data.

Reply:
renamed `AdminComment` and all related classes to `Comment`, to be less specific